api/src/controllers/settlements.rs

```diff
@@ -27,6 +36,29 @@ pub fn index(
     Ok(WebPayload::new(StatusCode::OK, payload))
 }
 
+pub fn create(
+    (connection, new_settlement, path, user): (
+        Connection,
+        Json<NewSettlementRequest>,
+        Path<PathParameters>,
+        AuthUser,
+    ),
+) -> Result<HttpResponse, BigNeonError> {
+    let connection = connection.get();
+    let organization = Organization::find(path.id, connection)?;
+    user.requires_scope_for_organization(Scopes::SettlementWrite, &organization, connection)?;
+    let new_settlement = Settlement::create(
+        organization.id,
+        new_settlement.start_time,
+        new_settlement.end_time,
+        SettlementStatus::PendingSettlement,
+        new_settlement.comment.clone(),
+        organization.settlement_type == SettlementTypes::PostEvent,
+    );
+    let settlement = new_settlement.commit(connection)?;
```

Comment:
Maybe add the user in here to pass to any domain events that are created. 
```suggestion
    let settlement = new_settlement.commit(user.user, connection)?;
```

Reply:
Sounds good, will do.

Reply:
All set,

```
select * from domain_events order by created_at desc limit 1;
                  id                  |        event_type         |     display_text     |                        event_data                        |  main_table   |               main_id                |         created_at         |         updated_at         |               user_id                | seq  | organization_id 
--------------------------------------+---------------------------+----------------------+----------------------------------------------------------+---------------+--------------------------------------+----------------------------+----------------------------+--------------------------------------+------+-----------------
 80fb9655-1ea9-42ab-b28f-c756d4c84e33 | SettlementReportProcessed | Settlement processed | {"settlement_id":"3232e6c7-bdd9-47cf-985f-939daff39046"} | Organizations | cf1fea01-8a9a-4218-8a47-7ef4670217cd | 2019-10-22 16:16:15.149872 | 2019-10-22 16:16:15.149872 | bff9c54a-fa3e-40e9-b15b-610a1a0c7d1d | 2899 | 
(1 row)
```

User is now passed in for the manual settlements so the associated domain event contains their ID.